app/src/main/java/com/android_academy/covid_19/repository/CollisionDataRepo.kt

```diff
@@ -0,0 +1,100 @@
+package com.android_academy.covid_19.repository
+
+import com.android_academy.covid_19.db.dao.CollisionLocationsDao
+import com.android_academy.covid_19.db.dao.RoomCollisionLocationEntity
+import com.android_academy.covid_19.network.service.InfectedLocationModel
+import com.android_academy.covid_19.providers.UserLocationModel
+import com.android_academy.covid_19.ui.notification.CodeOrangeNotificationManager
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.launch
+
+interface CollisionDataRepo {
+    fun onCollisionsFound(collidingUserLocations: List<Pair<UserLocationModel, InfectedLocationModel>>)
+    fun getCollisions(): Flow<List<RoomCollisionLocationEntity>>
+    fun deleteAll()
+}
+
+class CollisionDataRepoImpl(
+    private val notificationManager: CodeOrangeNotificationManager,
+    private val collisionLocationsDao: CollisionLocationsDao
+) : CollisionDataRepo {
+
+    private val scope = CoroutineScope(Dispatchers.IO)
+
+    override fun onCollisionsFound(foundCollidingUserLocations: List<Pair<UserLocationModel, InfectedLocationModel>>) {
+        scope.launch {
+            var isNewCollisions = false
+            val existingCollisionList = collisionLocationsDao.getCollisionLocationsList()
+
+            foundCollidingUserLocations.forEach { collisions ->
```

Comment:
I think that here should be the condition that if "existingsCollisionsList" is empty all the results should be saved instead of going one by one and also to avoid the comparison.

Reply:
Check addCollisionToStorage() function
If the list is empty is add it one by one? 
Why
Because event if you call addList in DAO its still doing it one by one so it's no difference